python-frontend/src/main/java/org/sonar/plugins/python/api/tree/OrPattern.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * SonarQube Python Plugin
+ * Copyright (C) 2011-2021 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.plugins.python.api.tree;
+
+import java.util.List;
+
+/**
+ * OR pattern
+ *
+ * <pre>
+ *   case "foo" | "bar": ...
+ * </pre>
+ *
+ * See https://docs.python.org/3/reference/compound_stmts.html#or-patterns
+ */
+public interface OrPattern extends Pattern {
+
+  List<Pattern> patterns();
+
+  List<Token> separators();
```

Comment:
do we need to have `separators` tokens in the interface?
I think they might be potentially useful for reporting issue locations,  however I don't know if we would ever need to access a specific separator. Do you have some other use case in mind?

Reply:
I don't know, I saw that [`ExpressionList`](https://github.com/SonarSource/sonar-python/blob/master/python-frontend/src/main/java/org/sonar/plugins/python/api/tree/ExpressionList.java) has a list of commas.
It could be useful if we want to raise an issue on the following code:
```
case 'foo' | 'foo': ...
```
What do you think?

Reply:
I think that regarding tokens we've not been super consisting in the past. (sometimes we did keep them in the API, sometimes we didn't).

Anyway I'm fine with keeping `separators()`